packages/components/src/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -81,6 +81,15 @@ export interface ConfirmationModalRef {
   show(props: Omit<DisplayAction, "type">): void;
 }
 
+interface KeyboardEventCompareratorOption {
+  readonly key: string;
+  readonly shiftKey?: boolean;
+  readonly ctrlKey?: boolean;
+  readonly altKey?: boolean;
+  readonly metaKey?: boolean;
+  [propName: string]: string | boolean | undefined;
+}
```

Comment:
I was thinking about this type with regards to the handler below I wonder how you would feel about something like this?

```suggestion
type SimpleKeyComparerator = string;

interface VerboseKeyComparerator {
  readonly key: SimpleKeyComparerator;
  readonly shiftKey?: boolean;
  readonly ctrlKey?: boolean;
  readonly altKey?: boolean;
  readonly metaKey?: boolean;
}

type KeyComparerator = XOR< VerboseKeyComparerator, SimpleKeyComparerator>
```